stories/assets/js/select.js

```diff
@@ -0,0 +1,161 @@
+      this.defocusItem(document.getElementById(this.optionSelectedId));
+      this.optionSelectedId = ev.currentTarget.id;
+      this.value = ev.currentTarget.dataset.value;
+
+      this.close();
+      this.changeSelectedOption();
+    });
+  }
+
+  changeSelectedOption() {
+    // Using this instead of innerText or innerHTML so trapFocus mutationObserver doesn't trigger
+    this.$buttonTrigger[0].firstChild.nodeValue = this.$selectOptions.find(`#${this.optionSelectedId}`).text();
+
+    this.$currentSelect.data('selected-value', this.value);
+    this.$currentSelect.trigger('change');
+  }
+
+  close() {
+    this.$selectOptions.removeClass(this.activeClass);
+    this.$currentSelect.removeClass(this.expandedClass);
+  }
+
+  setupFocus() {
+    if (this.optionSelectedId) {
+      return;
+    }
+
+    this.focusFirstItem();
+  }
+
+  focusFirstItem() {
+    const firstItem = this.$selectOptions[0].querySelector('[role="option"]');
+
+    if (firstItem) {
+      this.focusItem(firstItem);
+    }
+  }
+
+  focusItem(element) {
+    this.defocusItem(document.getElementById(this.optionSelectedId));
+
+    element.classList.add('focused');
+
+    this.$selectOptions.attr('aria-activedescendant', element.id);
+    this.optionSelectedId = element.id;
+    this.value = element.dataset.value;
+
+    if (this.$selectOptions[0].scrollHeight > this.$selectOptions[0].clientHeight) {
+      const scrollBottom = this.$selectOptions[0].clientHeight + this.$selectOptions[0].scrollTop;
+      const elementBottom = element.offsetTop + element.offsetHeight;
+      if (elementBottom > scrollBottom) {
+        this.$selectOptions[0].scrollTop = elementBottom - this.$selectOptions[0].clientHeight;
+      } else if (element.offsetTop < this.$selectOptions[0].scrollTop) {
+        this.$selectOptions[0].scrollTop = element.offsetTop;
+      }
+    }
+
+    this.changeSelectedOption();
+  }
+
+  defocusItem(element) {
+    if (!element) {
+      return;
+    }
+    element.classList.remove('focused');
+  }
+
+  checkKeyPress(evt) {
+    const key = evt.which || evt.keyCode;
+    let nextItem = document.getElementById(this.optionSelectedId);
+
+    switch (key) {
+      case window.UNDP.keyCode.UP:
+      case window.UNDP.keyCode.DOWN:
+        evt.preventDefault();
+
+        if (key === window.UNDP.keyCode.UP) {
+          nextItem = nextItem.previousElementSibling;
+        } else {
+          nextItem = nextItem.nextElementSibling;
+        }
+
+        if (nextItem) {
+          this.focusItem(nextItem);
+        }
+
+        break;
+      case window.UNDP.keyCode.SPACE:
+      case window.UNDP.keyCode.RETURN:
+        evt.preventDefault();
+        this.value = nextItem.dataset.value;
+
+        this.close();
+        this.changeSelectedOption();
+        break;
+      default:
+        break;
+    }
+  }
+}
+// const customSelect = () => {
+//  const selects = $('[data-multi-select]');
+//  select.map((items, i) => {
+//
+//  });
+// };
```

Comment:
@Ashish-srij Please remove this extra code

Reply:
Done